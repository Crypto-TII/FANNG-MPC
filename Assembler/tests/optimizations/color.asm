ldint r1, 1 # 2
ldint r2, 1 # 24
ldint r4, 1 # 27
ldi c6, 1
ldint r7, 5 # 32
subint r3, r1, r2 # 25
subint r5, r1, r4 # 28
stmci c6, r3 # 26
pushint r5 # 29
pushint r7 # 33
ldint r6, 1 # 31
ldint r8, 1 # 34
ldint r15, 1 # 63
jmpne r15, 0, -14 # 64
ldint r17, 1 # 66
subint r18, r8, r17 # 67
pushint r18 # 68
