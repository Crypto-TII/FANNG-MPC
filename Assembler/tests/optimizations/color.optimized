ldint r1, 1 # 2
ldint r2, 1 # 24
ldint r3, 1 # 27
ldi c0, 1
ldint r4, 5 # 32
ldint r0, 1 # 34
ldint r5, 1 # 63
subint r2, r1, r2 # 25
subint r1, r1, r3 # 28
stmci c0, r2 # 26
pushint r1 # 29
pushint r4 # 33
jmpne r5, 0, -13 # 64
ldint r1, 1 # 66
subint r1, r0, r1 # 67
pushint r1 # 68
