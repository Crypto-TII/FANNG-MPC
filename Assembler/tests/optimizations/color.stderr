warning: Write to register is never read
  --> tests/optimizations/color.asm:11:1
   |
11 | ldint r6, 1 # 31
   | ---------------- register 6 written to but will never be read from again
   |